dnf5-plugins/builddep_plugin/builddep.cpp

```diff
@@ -217,6 +217,16 @@ bool BuildDepCommand::add_from_pkg(
     }
 }
 
+std::string escape_glob(const std::string & in) {
+    // Escape fnmatch glob characters in a string
+    std::string out = std::regex_replace(in, std::regex("\\\\"), "\\\\");
+    out = std::regex_replace(out, std::regex("\\["), "\\[");
+    out = std::regex_replace(out, std::regex("\\]"), "\\]");
+    out = std::regex_replace(out, std::regex("\\*"), "\\*");
+    out = std::regex_replace(out, std::regex("\\?"), "\\?");
```

Comment:
Such strings are going to stay in the "glob pattern" category because of the [simple check](https://github.com/rpm-software-management/dnf5/blob/7ea44d9271ecdf391f1bb0f4661f73b48fdaa6d8/include/libdnf5/utils/patterns.hpp#L29-L31).  Is that OK for `is_glob_pattern` callers?

Reply:
I assume yes... just that DNF5/libdnf goes through more expensive code paths...  but long-term, #1085 looks like semantically more correct.  Perhaps "the" issue is about to be fixed in two phases?

Reply:
Yes, the code path is a bit more expensive, but anyway I think the escaped strings _need_ to be handled as globs or else the escaping backslashes (`\[`, etc) will be treated as literal and not as escaping the following glob character.

#1085 is definitely the better fix in the long term, but I think both patches are small and tidy enough that we can do this for now, and possibly revert it in favor of #1085 later.

Reply:
Regular expressions seem to be a bit of overkill here. How about doing the escaping in a single loop?
```
  std::string out;
  for (const auto ch : str) {
    if (ch == '*' || ch == '?' || ch == '[' || ch == ']') {
      out += '\\';
    }
    out += ch;
  }
  return out;
```
But I'm not blocking the merge on it if you prefer regex.

Reply:
Done, that's saner and I think still plenty readable.